mantidproject/mantid
# Shorten arrays with integer lists

Improves list handling for long lists in properties, integer separated lists such as 1,2,3 will collapse into 1-3.

**To test:**


1. Check unit and system tests pass, there are new unit tests for this.
1. Open Mantidplot
1. Set the log windows to display information logs (right click the window)
1. Run this script (it's easier in chunks)
1. Look at the values in the Workspace History GUI as well.
1. Check that generated scripts still work as expected (they may include list shortening, but that should still work).
``` python 
dataX = range(100)
dataY = range(100)
dataE = range(100)

# The workspace will be named "dataWS"
dataWS = CreateWorkspace(DataX=dataX, DataY=dataY, DataE=dataE, NSpec=1,UnitX="Wavelength")

'''Log entries like this (truncation, but not list shortening as these are double properties)
  Name: DataX, Value: 0,1,2,3,4,5,6,7,8 ... 94,95,96,97,98,99, Default?: No, Direction: Input
  Name: DataY, Value: 0,1,2,3,4,5,6,7,8 ... 94,95,96,97,98,99, Default?: No, Direction: Input
  Name: DataE, Value: 0,1,2,3,4,5,6,7,8 ... 94,95,96,97,98,99, Default?: No, Direction: Input
  '''


speclist = []
speclist += range(1,100)
CreateSampleWorkspace(OutputWorkspace='base', Function='Powder Diffraction')
MaskDetectors(Workspace='base', SpectraList=speclist)
'''Log entry like this (list shortening, no truncation)
  Name: SpectraList, Value: 1-99, Default?: No, Direction: Input
  '''

speclist = []
speclist += range(1,3)
speclist += range(4,50,2)
speclist += range(60,100)
CreateSampleWorkspace(OutputWorkspace='base', Function='Powder Diffraction')
MaskDetectors(Workspace='base', SpectraList=speclist)

'''Log entry like this (list shortening, and truncation)
  Name: SpectraList, Value: 1-2,4,6,8,10,12,1 ... 42,44,46,48,60-99, Default?: No, Direction: Input
  '''
```

**Python interface test**
``` python
speclist = []
speclist += range(1,3)
speclist += range(4,50,2)
speclist += range(60,100)
base = CreateSampleWorkspace(OutputWorkspace='base', Function='Powder Diffraction')
MaskDetectors(Workspace='base', SpectraList=speclist)


alg = base.getHistory().lastAlgorithm()
p= alg.getProperty("SpectraList")


print "List collapsing, no length limit",p.valueAsPrettyStr()
print "No list collapsing, no length limit",p.valueAsPrettyStr(0,False)
print "List collapsing, length limit 40",p.valueAsPrettyStr(40,True)
print "No list collapsing, length limit 40",p.valueAsPrettyStr(40,False)
```

closes #15164



**Release Notes** 

docs/source/release/ui.rst


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
